### ISSUE TITLE
Add a unique identifier to the article content container

### ISSUE THREAD

**Issue by IceQ1337** [WoltLab/WCF#4887]
Apologies if this should better be posted in the suggestions forum, but due to the small scope of the suggested change, I just wanted to quickly open an issue instead.

At the moment, the [container](https://github.com/WoltLab/WCF/blob/master/com.woltlab.wcf/templates/article.tpl#L126) that contains the article image, the teaser and the actual content of the article does not have a unique identifier.

Unfortunately, applying CSS rules to this container is often not that easy or consistent, especially since the position/order of this element in the DOM cannot be guaranteed. This can be caused, for example, by `.userNotice` or `.contentInteraction` containers, each of which is dynamically either present or absent on a case-by-case basis.

I don't expect browsers to add support for complex CSS selectors (e.g. [nth-child-of](https://caniuse.com/css-nth-child-of)) anytime soon, so I therefore have somewhat of a last-minute request for adding a simple class like `.articleContentContainer` for this container in WSC 5.5, but maybe this can even be applied retrospectively to 5.3 and 5.4, because not updating custom templates probably wouldn't do any harm.

**Comment by IceQ1337** [comment]
Of course. Arguably the simplest example would probably be ordering the very containers that this one contains, such as moving the article image above the teaser. Simply changing the `order` on these elements has no effect because their not a flex/grid element.

It would require applying the following CSS rules to this particular parent container first:
```CSS
display: flex;
flex-direction: column;
```

Using selectors like `#content div:nth-child(X)` or [`.articleContentHeader + .section`](https://www.woltlab.com/community/thread/281365-einleitungstext-unter-dem-bild/?postID=1783035#post1783035) will immediately fail if a `.userNotice` is created which leads to inconsistencies in the presentation of the content.

A unique identifier would also allow style authors to target the container as a whole excluding the header and footer, but I don't have a specific use case for this that isn't already possible to accomplish through the nested classes you mentioned.

**Comment by dtdesign** [comment]
```css
body[data-page-identifier="com.woltlab.wcf.Article"] #content > .section:not(.articleNavigation) {
  /* yada yada yada */
}
```

It‘s arguably not the most elegant solution, but it gets the job done?

**Comment by IceQ1337** [comment]
This selector is at least more definite than the ones I mentioned, but would also apply to the related articles section or third-party content added through template listeners, for example. While adding additional exceptions isn't a big deal, stacking up `:not()` selectors might not be the best/most elegant approach. In terms of just the WSC, it's probably manageable, but I also think that an extra class would not suddenly result in an excessive DOM size. It will however, make the direct access to this element much easier and absolutely definite.

**Comment by dtdesign** [comment]
Fair enough, I have added the CSS class `.articleContainer` that will be shipped with 5.5.0.
